packages/amplify-category-function/src/commands/function/remove.ts

```diff
@@ -33,12 +38,23 @@ module.exports = {
       resourceName = response.resourceName;
     }
 
+    let hasSecrets = false;
+
+    const resourceNameCallback = async (resourceName: string) => {
+      hasSecrets = getLocalFunctionSecretNames(resourceName).length > 0;
+    };
+
     return amplify
-      .removeResource(context, categoryName, resourceName)
-      .then((resource: { service: string; resourceName: string }) => {
+      .removeResource(context, categoryName, resourceName, undefined, resourceNameCallback)
```

Comment:
nit: Would it be better to await the promises here as opposed to callbacks?

Reply:
The problem is I need to run this logic after the resourceName is resolved but before any resources are actually removed. Because both of these things happen inside of the amplify.removeResource() method, the only way to do that (without more extensive refactoring) is to inject a callback into that function